dev/diagrams/on-app-booting-hook.md

```diff
@@ -0,0 +1,60 @@
+# On App Booting Hook
+
+<!-- TOC -->
+
+- [Use Is App Booted Hook](#use-is-app-booted-hook)
+  - [Overview](#overview)
+
+<!-- /TOC -->
+
+## Overview
+This is a basic activity diagram for the useIsAppBootedHook.
+
+
+```plantuml
+
+@startuml
+
+title useIsAppBootedHook()
+
+participant "End User" as user
+
+box "FE" #Lightblue
+  participant "Component" as fe
+  participant "Hook: useIsAppBooted" as hook
+	participant "UseAuthPrepHook" as authPrep
+	participant "Recoil" as recoil
+box
+box "API" #Lightgreen
+  participant "API" as api
+box
+
+activate user
+	user -> fe: open app
+	activate fe
+		fe -> hook: useIsAppBooted
+		activate hook
+			hook -> authPrep: onGetAuthPrep
+			activate authPrep
+				break if there is no auth prep or the user is not signed in
+					hook <- authPrep: throw new Error('Not Singed In')
+					fe <- hook: router.push(DEFAULT_MAIN_APP_PAGE)
+				end break
+				hook <- authPrep: handleSignOutApp()
+			deactivate authPrep
+			hook -> recoil: useEffect()
+			activate recoil
+				break if isBooted is true
+			  	hook <- recoil: Returns onAppBooting()
+				end break
+			deactivate recoil
```

Comment:
```suggestion
```
Can you explain what `useEffect()` is by 1/23?
https://react.dev/reference/react/useEffect

Reply:
useEffect() is used to achieve side effects within a component.
It is executed after the component has been drawn and can monitor changes in the state of certain values.

Reply:
nice.